src/wallet/scriptpubkeyman.cpp

```diff
@@ -1174,7 +1174,7 @@ bool DescriptorScriptPubKeyMan::CanGetAddresses(bool internal) const
     LOCK(cs_desc_man);
     return m_wallet_descriptor.descriptor->IsSingleType() &&
            m_wallet_descriptor.descriptor->IsRange() &&
-           (HavePrivateKeys() || m_wallet_descriptor.next_index < m_wallet_descriptor.range_end);
+           (HavePrivateKeys() || m_wallet_descriptor.next_index < m_wallet_descriptor.range_end || m_wallet_descriptor.descriptor->CanSelfExpand());
```

Comment:
In 52583d3269cc747ff58b6d0b8c213a1ac521aeb3 "wallet: Use Descriptor::CanSelfExpand() in CanGetAddresses()": would be nice to have to test that illustrates a case where this matters

Reply:
This PR does that.

Without this PR, that test would not be possible as it is inherently contradictory. We want to test that we are able to get addresses from something for which we are not able to get addresses for in a watchonly wallet. The only way a watchonly wallet could have those addresses are if it had private keys and derived the keys already. But a watchonly wallet cannot have private keys. This situation is literally impossible to test prior to this PR.

The only reason such a test works with this PR is because we can bring along the descriptor caches when making the watchonly wallet, so the pubkeys are already derived.

Reply:
In 9317ea0b1cfa2f31dd78c3ec58c6f6eeedf9ba20 _wallet: Use Descriptor::CanSelfExpand() in CanGetAddresses()_:

The explanation makes sense, but I can drop `|| m_wallet_descriptor.descriptor->CanSelfExpand()` (after 3ebf3f7b58aade3da20b50bf69c5ef8fe5ac2f0f _test: Test for exportwatchonlywallet_) without breaking `wallet_exported_watchonly.py`.